### DISCUSSION TITLE
Create PDF using paragraphs and tables

### DISCUSSION THREAD

**Discussion by sascha-schwegelbauer** [UglyToad/PdfPig#408] [Q&A]
Hi!

Do any plans exist to enhance PdfPig with layout-features for creation of new PDF-Files like paragraphs and tables?
Thanks for a short reply.

Regards
Sascha

**Comment by EliotJones** [comment]
Hi @sascha-schwegelbauer,

No plans currently. If I was more productive I'd consider building a commercial product around this since it's not an easy problem to solve (well), for example just implementing good text-wrapping for paragraphs has some complexity: https://en.wikipedia.org/wiki/Line_wrap_and_word_wrap

For now the creation API exposes enough data/methods (such as the ability to measure text before placement) that consumers can hopefully build such implementations themselves.

**Reply by sascha-schwegelbauer** [reply]
> Hi @sascha-schwegelbauer,
> 
> No plans currently. If I was more productive I'd consider building a commercial product around this since it's not an easy problem to solve (well), for example just implementing good text-wrapping for paragraphs has some complexity: https://en.wikipedia.org/wiki/Line_wrap_and_word_wrap
> 
> For now the creation API exposes enough data/methods (such as the ability to measure text before placement) that consumers can hopefully build such implementations themselves.

Hi @EliotJones ,

thanks for the fast reply! :-)
I fully ack - it's really not simple - the whole layouting has to be calculated and projected to pdf using the measure-text-api.

Another question: are any plans for supporting PDF-A/3 (regarding attachments)
